src/main/java/org/generationcp/middleware/api/germplasm/GermplasmNameService.java

```diff
@@ -0,0 +1,20 @@
+package org.generationcp.middleware.api.germplasm;
+
+import org.generationcp.middleware.domain.germplasm.GermplasmNameRequestDto;
+import org.generationcp.middleware.pojos.Name;
+import org.generationcp.middleware.pojos.UserDefinedField;
+
+public interface GermplasmNameService {
+
+	Name getNameByNameId(Integer nameId);
+
+	Name getPreferredName(Integer gid);
```

Comment:
Remove this function and create it in NameDao directly

Reply:
done.